docs/meshstack.customer-group-sync.md

```diff
@@ -3,142 +3,192 @@ id: meshstack.customer-group-sync
     -}
-      { mesh-object : Text, fields : List Field, tags : List Field }
+      < Static : StaticAttributeTransformation
+      | Regex : RegexAttributeTransformation
+      >
 ```
 <!--END_DOCUSAURUS_CODE_TABS-->
 
-The Field configuration mentioned above has two possible forms. A basic form known as `BaseField` and a Field that performs
-transformations on the matched attribute, called the `RegexField`. The `RegexField` is an extension of the `BaseField`.
+#### StaticAttributeTransformation
 
-<!--snippet:mesh.identityconnector.ldap.baseField#type-->
+<!--snippet:mesh.identityconnector.ldap.staticAttributeTransformation#type-->
 
 
 <!--DOCUSAURUS_CODE_TABS-->
 <!--Dhall Type-->
 ```dhall
-let BaseField =
+let StaticAttributeTransformation =
     {-
-      transformationType:
-          One of "static" or "regex". Static transformation will simply take the LDAP attribute value.
-          The regex transformation will perform extra operations on the matched value.
-
-      field:
-          The meshObject field that should be assigned the result of this field mapping.
-          For example, "metadata.name"
+      A StaticAttributeTransformation simply takes an attribute and applies an optional postProcessor on it.
 
       attribute:
           The LDAP attribute that should be processed. For example "cn"
 
       postProcessor:
           Any post processing function that should be run on the mapped value. Can be one of UPPERCASE or LOWERCASE
     -}
-      { transformationType : Text
-      , field : Text
-      , attribute : Text
-      , postProcessor : Optional Text
-      }
+      { attribute : Text, postProcessor : Optional PostProcessor }
+```
+<!--Example-->
+```dhall
+let example =
+      { attribute = "cn", postProcessor = Some PostProcessor.LOWERCASE }
 ```
 <!--END_DOCUSAURUS_CODE_TABS-->
 
-<!--snippet:mesh.identityconnector.ldap.regexField#type-->
+#### RegexAttributeTransformation
+
+<!--snippet:mesh.identityconnector.ldap.regexAttributeTransformation#type-->
 
 
 <!--DOCUSAURUS_CODE_TABS-->
 <!--Dhall Type-->
 ```dhall
-let RegexField =
+let RegexAttributeTransformation =
     {-
+     A RegexAttributeTransformation takes an attribute and matches the attribute against a list of regular expressions
+     until a match is found. If a match is found and the rule defines a value, the value will be used. If a value is not
+     set, the first matching group in the regular expression is used and the optional template is applied. If no
+     regular expressions match, the otherwise value will be assigned to the meshObject field.
+
       rules:
           A list of regex rules that the LDAP attributes will be tested against for a match.
           The matching is performed sequentially until a match is found.
+          A single rule has the following structure
+          value:
+              This is an optional parameter. If it is defined, if the LDAP attribute matches the
```

Comment:
```suggestion
              This is an optional parameter. It is defined, if the LDAP attribute matches the
```
```suggestion
              This is an optional parameter. If it is defined, if the LDAP attribute matches the
```

Reply:
Changed the wording to `If it is defined and if  the LDAP ....`